### PR TITLE
RATIS-1735. Fix an spelling mistake in annotation

### PR DESCRIPTION
**Pull request by bzheng888**
See https://issues.apache.org/jira/browse/RATIS-1735

### CODE REVIEW

**Comment by bzheng888** [comment]
@codings-dan Please take a look, thanks!

**Review by codings-dan — APPROVED** [review]
@bzheng888 Welcome to ratis community, the change looks good, +1

File: ratis-common/src/main/java/org/apache/ratis/util/LifeCycle.java [hunk 1]
```diff
@@ -223,7 +223,7 @@ public State transitionAndGet(UnaryOperator<State> operator) {
    * If the current state is equal to the specified from state,
    * then transition to the give to state; otherwise, make no change.
    *
-   * @return true iff the current state is equal to the specified from state.
+   * @return true if the current state is equal to the specified from state.
```
Comment (szetszwo) [inline ratis-common/src/main/java/org/apache/ratis/util/LifeCycle.java:226]: Thanks for checking the javadoc!

This is not a typo iff means "if and only if"; see https://www.merriam-webster.com/dictionary/iff

Comment (bzheng888) [reply ratis-common/src/main/java/org/apache/ratis/util/LifeCycle.java:226]: Amazing,  I didn't know this usage before

Comment (codings-dan) [reply ratis-common/src/main/java/org/apache/ratis/util/LifeCycle.java:226]: > This is not a typo iff means "if and only if";

Thanks for the info, I also not hear of this usage

**Pull request closed without merging**